### PR TITLE
v2 publish endpoint needs to be read from v3 index.json

### PR DESCRIPTION
**Pull request by deepakaravindr**
Fixes https://github.com/NuGet/Home/issues/1202

Using PushCommandResource to get the push endpoint for v3 package source.
With this change, we can push packages to a v3 server, in that,
to the v2 endpoint, if one is available in v3's index.json

@yishaigalatzer @emgarten @pranavkm @feiling @johnataylor

### CODE REVIEW

File: src/NuGet.CommandLine/Commands/PushCommand.cs [hunk 1]
```diff
@@ -53,6 +57,27 @@ public override void ExecuteCommand()
...
+        {
+            var packageSource = new Configuration.PackageSource(source);
+
+            var sourceRepositoryProvider = new SourceRepositoryProvider(SourceProvider,
+                Enumerable.Concat(
+                    Protocol.Core.v2.FactoryExtensionsV2.GetCoreV2(Repository.Provider),
+                    Protocol.Core.v3.FactoryExtensionsV2.GetCoreV3(Repository.Provider)));
+
+            var sourceRepository = sourceRepositoryProvider.CreateRepository(packageSource);
+            var pushCommandResource = await sourceRepository.GetResourceAsync<PushCommandResource>();
```
Comment (pranavkm) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:70]: What's the experience for this like? Would you have to run:
`nuget.exe push -Source https://api.nuget.org/index.json {Package}`

File: test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs [hunk 2]
```diff
@@ -159,7 +160,7 @@ public void PushCommand_PushToServer()
                     server.Start();
 
                     // Act
-                    string[] args = new string[] { "push", packageFileName, "-Source", server.Uri + "push" };
+                    string[] args = new string[] { "--debug", "push", packageFileName, "-Source", server.Uri + "push" };
```
Comment (MeniZalzman) [inline test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs:163]: remove debug

File: src/NuGet.CommandLine/Commands/PushCommand.cs (hunk 1 above)
Comment (deepakaravindr) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:70]: Yes. With this change, that will work too

Comment (pranavkm) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:70]: That seems weird though - attempting to push to a json file?

**Comment by emgarten** [comment]
:shipit:

File: src/NuGet.CommandLine/Commands/PushCommand.cs (hunk 1 above)
Comment (deepakaravindr) [inline src/NuGet.CommandLine/Commands/PushCommand.cs:70]: But, you are not publishing to a json file. You are pushing to a v3 package source. The index.json tells you what the publish endpoint is

**Comment by MeniZalzman** [comment]
:shipit:

File: test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs (hunk 2 above)
Comment (deepakaravindr) [inline test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs:163]: fixed

**Comment by deepakaravindr** [comment]
Merged with https://github.com/NuGet/NuGet.PackageManagement/commit/70cc0e4236ed1438742681996974e49252c39ba3

**Pull request closed without merging**